util.py

```diff
@@ -0,0 +1,193 @@
+    for pattern in patterns: sentence = sentence.replace(pattern[0], pattern[1])
+    punctuation = ["!),.:;?]}", "([{"]
+    old = ""
+
+    while (sentence != old):
+        old = deepcopy(sentence)
+        sentence = sentence.replace("  ", " ")
+        for punct in punctuation[0]:
+            for space in string.whitespace:
+                sentence = sentence.replace(str(space+punct), str(punct))
+            sentence = sentence.replace(str(" "+punct), str(punct))
+            sentence = sentence.replace(str(punct), str(punct+" "))
+            for puncta in punctuation[0]:
+                sentence = sentence.replace(str(puncta+" "+punct), str(puncta+punct))
+                sentence = sentence.replace(str(punct+" "+puncta), str(punct+puncta))
+
+        for punct in punctuation[1]:
+            for space in string.whitespace:
+                sentence = sentence.replace(str(punct+space), str(punct))
+            sentence = sentence.replace(str(punct+" "), str(punct))
+            sentence = sentence.replace(str(punct), str(" "+punct))
+            for puncta in punctuation[0]:
+                sentence = sentence.replace(str(puncta+" "+punct), str(puncta+punct))
+                sentence = sentence.replace(str(punct+" "+puncta), str(punct+puncta))
+
+        sentence = sentence.replace("  ", " ")
+        if (sentence[len(sentence)-1] == " "): sentence = sentence[:len(sentence)-1]
+        if (sentence[0] == " "): sentence = sentence[1:]
+
+    old = ""
+    for space in string.whitespace: sentence = sentence.replace(str(space), str(" "))
+    while (sentence != old):
+        old = deepcopy(sentence)
+        sentence = sentence.replace("  ", " ")
+
+    for pattern in patterns: sentence = sentence.replace(pattern[1], pattern[0])
+    return sentence
+
+def format_jsonfile (dialogs, filename, sufix):
+    with open(filename.replace(".json", sufix+".json"), 'w', encoding='utf-8') as f:
+        f.write(json.dumps(dialogs, indent=2, ensure_ascii=False))
+
+def order_dialog(dialog):
+    result = {}
+    result["id"] = dialog["id"]
+    result["dialog_domain"] = dialog["dialog_domain"]
+    result["turns"] = []
+
+    for turn in dialog["turns"]:
+        new_turn = {}
+        new_turn["speaker"] = turn["speaker"]
+        new_turn["utterance"] = turn["utterance"]
+        new_turn["utterance_delex"] = turn["utterance_delex"]
+        if ("intent" in turn.keys()): new_turn["intent"] = turn["intent"]
+        if ("action" in turn.keys()): new_turn["action"] = turn["action"]
+        new_turn["slot-values"] = turn["slot-values"]
+        new_turn["turn-num"] = turn["turn-num"]
+        result["turns"].append(new_turn)
+    return result
+
+def organizeturn (turn, delex):
+    new_turn = turn.copy()
+    new_turn["utterance"] = delex
+    new_turn["utterance_delex"] = delex
+    new_turn["slot-values"] = turn["slot-values"].copy()
+    return new_turn
+
+def strTimeProp (start, end, format, prop):
```

Comment:
python uses snake case, not camelcase, so the name should be str_time_prop(...)

Reply:
Solved.